jmespath-core/src/main/java/io/burt/jmespath/function/NotNullFunction.java

```diff
@@ -0,0 +1,21 @@
+package io.burt.jmespath.function;
+
+import java.util.List;
+
+import io.burt.jmespath.Adapter;
+
+public class NotNullFunction extends JmesPathFunction {
+  public NotNullFunction() {
+    super(1, Integer.MAX_VALUE);
+  }
+
+  @Override
+  protected <T> T internalCall(Adapter<T> adapter, List<ExpressionOrValue<T>> arguments) {
+    for (ExpressionOrValue<T> x : arguments) {
+      if (!adapter.isNull(x.value())) {
+        return x.value();
+      }
```

Comment:
Yes, and I think that is a problem in more places than this, and I would like to move the type checking up into `JmesPathFunction` and have the subclasses declare their type expectations. Not sure when I should put in that work though, and if it's worth it.